packages/main/src/plugin/kubernetes/contexts-manager-experimental.ts

```diff
@@ -16,12 +16,57 @@
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
 
-import type { KubeConfig, KubernetesObject } from '@kubernetes/client-node';
+import type { Context, KubernetesObject } from '@kubernetes/client-node';
+import { KubeConfig } from '@kubernetes/client-node';
 
 import type { ContextGeneralState, ResourceName } from '/@api/kubernetes-contexts-states.js';
 
+import { ContextsConnectivityRegistry } from './contexts-connectivity-registry.js';
+import { HealthChecker } from './health-checker.js';
+
+const HEALTH_CHECK_TIMEOUT_MS = 5_000;
+
 export class ContextsManagerExperimental {
-  async update(_kubeconfig: KubeConfig): Promise<void> {}
+  // kubeConfigCheck is used by `update` to check if the received kubeconfig has changed
+  // its value must not be used except for detecting this change
+  // Methods of this class should receive a kubeconfig from `update`
+  #kubeConfigCheck = JSON.stringify(new KubeConfig());
```

Comment:
I would really have a preference to avoid doing that and have a `compare(a: KubeConfig, b: KubeConfig)` method.

Reply:
This would need a deep comparison, and some fields are of type any (`User.exec?: any;` for example). I'm not sure it would be easy to find/write a robust-enough and efficient `compare` implementation.

I think that it could be a follow-up optimization

Reply:
AFAIK we have events monitoring the kubeconfig file.

kubeconfig file modified = we receive a event, so I'm wondering why we would need a comparison there ?